### DISCUSSION TITLE
getAnalogResolution() in DxCore and megaTinyCore

### DISCUSSION THREAD

**Discussion by Warren52** [SpenceKonde/megaTinyCore#1004] [Q&A]
So I am taking analog readings from a 10K Pot with one end tied through a 1K resistor to 5VDC and the other end to Ground.
I am putting code together that I want to work on both an AVR128DB28 (using DxCore) and an ATtiny 3224 (using metaTinyCore).

I just finished getting the DxCore code working nicely and moved it to my ATtiny 3224 / megaTinyCore setup and have hit a snag.
With the DxCore I have a working getAnalogResolution() function but the compile on megaTinyCore reports: undefined reference to `getAnalogReadResolution'

Your megaTinyCore https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/extras/Ref_Analog.md document includes documentation for the getAnalogReferece() function and I have searched both DxCore and megaTinyCore code finding BOTH have the function definition in their respective Arduino.h files but there is no body to be found in megaTinyCore.

My code isn't really rocket science:
if(getAnalogResolution() != 12)
    analogReadResolution(12).

So:
 - Is this just an oversight and the body will eventually be added to megaTinyCore or are there some other plans?
 - If there are other plans, what common code should be used to perform getAnalogResolution() for DxCore and megaTinyCore?

**Comment by Warren52** [comment]
Sorry - forgot to mention - DxCore 1.5.8, megaTinyCore 2.6.8

**Comment by SpenceKonde** [comment]
converted to issue #1006 

note that it is getAnalogReadResolution()
